xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -123,14 +128,44 @@ bool CVAAPIContext::EnsureContext(CVAAPIContext **ctx, CDecoder *decoder)
   return true;
 }
 
+namespace
+{
+int find_open_render_node_fd()
```

Comment:
You agree that vaapi processing is done by gpu, do you? Assume we have 2 cards: gpu1 and gpu2. We would also have two render nodes. Still agree? Now vaapi processing can either be on gpu1 or gpu2, right? Assume you select render node for device gpu1 but our display in on gpu2.
So vaapi decoding is done on gpu1 but we want to display the pic on display connected to gpu2. Does this sound as a reasonable scenario for you?